Add new Understanding Resource Timing documentation.

Garbee opened the pull request:
Adding a new page specifically for resource timing. Moving the data from the existing network page to it and expanding upon it.

Sorry @kaycebasques, you know I can't test locally. So please let me know what is needed to make sure this is linked properly from places like the DevTools landing page's Network tab. If it isn't automatically done.

crhym3 commented:
Hey @Garbee could you rebase this PR on master? The CI should now be able to stage every PR at a unique URL to browse. Just wanted to test it out with a real PR like yours.

Garbee commented:
Sure thing, rebasing now.

Garbee commented:
> [21:45:08] Finished 'build:staging' after 2.02 min
> $ tools/dump-ci-state.sh > .state
> /bin/sh: 31: tools/dump-ci-state.sh: not found
> [info] build failed (exit code 127)

Looks like a script isn't being referenced properly.

crhym3 commented:
Hm.. that's weird, looks like it didn't rebase. The [drone.yml](https://github.com/google/WebFundamentals/blob/4184cf8c0df94b76f24808364d88c29306f91187/.drone.yml) in this PR doesn't match what's in the master at the moment: https://github.com/google/WebFundamentals/blob/master/.drone.yml

Garbee commented:
Ah, just checked my git log where I'm editing. The pull from upstream didn't get the last couple merges. Repull and rebase again fixed it.

crhym3 commented:
Cool. This should now be browsable at https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/

Garbee commented:
Wahoo! That is *fantastic* work @crhym3! Very useful for people like me who are unable to build locally!

Thank you so much. :smiley:

crhym3 commented:
Glad you liked it. Maybe I should do the same thing for getmdl.io.

Garbee commented:
Let's have a discussion with Sergio and try to figure out our V2 plans a bit first before jumping into that with MDL. Things may change quite a bit with out setup in V2 and V1 work is winding down. I don't want anyone adding time into stuff right now for it to need to be heavily modified shortly.

Garbee commented:
[Hosted version](https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/tools/chrome-devtools/profile/network-performance/understanding-resource-timing?hl=en) of the primary doc update. Linked perfectly from the higher-up page.

Only thing that needs doing is a link in the main carousel.

Garbee commented:
Just added  a link to the doc on the homepage panel overview tabs.

That should be all the coverage here.

crhym3 commented:
@Garbee looks like your contributor photo is linked with incorrect extension jpg on https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/tools/chrome-devtools/profile/network-performance/understanding-resource-timing?hl=en.

The existing file ends with `.png`. It would still generate incorrect link because of an unrelated "absolute URL" setting but I fixed it in https://github.com/google/WebFundamentals/commit/753657e0c4341e61459e8d37bb6fc2ec7163cf1b.

Garbee commented:
I thought I replaced it in this branch with a jpeg... I'll double check
tomorrow.
On Feb 11, 2016 6:01 PM, "alex" <notifications@github.com> wrote:

> @Garbee <https://github.com/Garbee> looks like your contributor photo is
> linked with incorrect extension jpg on
> https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/tools/chrome-devtools/profile/network-performance/understanding-resource-timing?hl=en
> .
>
> The existing file ends with .png. It would still generate incorrect link
> because of an unrelated "absolute URL" setting but I fixed it in 753657e
> <https://github.com/google/WebFundamentals/commit/753657e0c4341e61459e8d37bb6fc2ec7163cf1b>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/WebFundamentals/pull/2516#issuecomment-183093433>
> .
>

Garbee commented:
Yea, I updated the png to a jpg in this PR. Rebasing again to see if the pr sub updates...

crhym3 commented:
Cool. It'll work now. It's just those images refer to the master build which doesn't have your image yet: https://web-central.appspot.com/web/imgs/contributors/jonathangarbee.jpg

but if you replace the URL with this PR's, it shows up: https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/imgs/contributors/jonathangarbee.jpg

Garbee commented:
Looks like the pr stage is not updating with my last update (made sub-headings `<small>` instead of doing actual low-level headings since we appear to be capped at 3 levels max.) I'm going to restart drone manually and see if that does it.

crhym3 commented:
Can you try in incognito window? It might be cached...

Garbee commented:
Not a cache problem. Manual rebuild and the [stage page](https://pr-2516-dot-weasel-dot-web-central.appspot.com/web/tools/chrome-devtools/profile/network-performance/understanding-resource-timing?hl=en) still shows `#### AKA: lots of green`  for example instead of nice small elements.

Garbee commented:
Just made an actual content edit. Let's see if that gets pushed.

crhym3 commented:
@Garbee checking... the actual content I'm seeing in the bucket is indeed `<small>AKA: lots of green</small>`.

crhym3 commented:
ok yeah, it's a bug. something I forgot. the app which serves pr builds caches responses and never purges the cache. fixing it...

crhym3 commented:
this should be fixed now.

Garbee commented:
Yup, latest content is showing. Thanks!

petele commented:
@kaycebasques - is this good to merge?

kaycebasques commented:
@garbee @petele I'll review tomorrow

petele commented:
@kaycebasques - ping?

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+{% include shared/toc.liquid %}
+
+{% include shared/takeaway.liquid list=page.key-takeaways.emulate-network %}
```
Your `TL;DR` is not displaying because this should be `list=page.key-takeaways.timing`

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+As they are retrieved over the network, resources have distinct lifecycles expressed in terms of resource timing.
+The network panel uses the same [Resource Timing API](http://www.w3.org/TR/resource-timing) that is available to application developers.
+
+The Resource Timing API provides a rich level of detail about each individual assets time to be received.
```
s/assets/asset's/

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+All network requests are considered resources.
+As they are retrieved over the network, resources have distinct lifecycles expressed in terms of resource timing.
+The network panel uses the same [Resource Timing API](http://www.w3.org/TR/resource-timing) that is available to application developers.
```
s/network panel/Network Panel/

Capitalize "Network Panel" throughout the document.

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+To view the full timing information for a given entry of the network panel you have three options.
+
+1. You may hover the timing graph under the timeline column. This will present a popup that shows the full timing data.
```
Drop the "You may" here and in the items below. `Hover over the timing graph under the timeline column.`

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+        The request was put on hold to wait for an unavailable TCP socket that's about to free up.
+        </li>
+        <li>
+        The request was put on hold because the browser only allows [six TCP connections](https://code.google.com/p/chromium/issues/detail?id=12066) per origin on HTTP 1.
```
This link isn't working. I don't think that Markdown works within HTML.

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+<dl>
+
+  <dt class="queued"><strong>Queuing</strong></dt>
```
I like what you did here with the color coding.

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+![Stalled series of requests](imgs/stalled-request-series.png)
+
+To fix this problem for traditional HTTP 1 traffic, you would need to implement domain sharding.
```
link off to a domain sharding article here

kaycebasques commented on src/content/en/tools/chrome-devtools/profile/network-performance/understanding-resource-timing.markdown:
```diff
@@ -0,0 +1,222 @@
+
+![High TTFB Indicator](imgs/indicator-of-high-ttfb.png)
+
+A slow time to first byte is recognized by a high waiting time.
```
Put TTFB in parentheses after `A slow time to first byte` so that people know the acronym.

`A slow time to first byte (TTFB) ...`

kaycebasques commented:
Great work @Garbee. I left some comments. Fix and those and I'll merge. After that I'm going to go through and do a proofread to tighten up the writing. I'll ping you when I do that.

Garbee commented:
Ok, I'll get these edits in at some point today.

Garbee commented:
All the issue should be fixed.

Thanks for pointing out to remove the 'You may' and 'You can' in the list. It reads much better.
